crates/alpen-ee/common/src/types/ee_update.rs

```diff
@@ -0,0 +1,87 @@
+use std::iter;
+
+use bitcoin_bosd::Descriptor;
+use strata_acct_types::Hash;
+use strata_identifiers::L1BlockCommitment;
+
+/// Unique identifier for an EeUpdate
+#[derive(Debug)]
+pub struct EeUpdateId {
+    pub prev_block: Hash,
+    pub last_block: Hash,
+}
+
+/// DA related data relevant to EeUpdate
+#[derive(Debug)]
+pub struct L1DaRef {
+    pub block: L1BlockCommitment,
+    pub txns: Vec<Descriptor>,
+    // inclusion merkle proof ?
+}
+
+/// EeUpdate lifecycle states
+#[derive(Debug)]
+pub enum EeUpdateStatus {
+    /// Newly created
+    Init,
+    /// DA started, waiting for block
+    DaPending { txids: Vec<Descriptor> },
+    /// DA complete
+    DaComplete { da: Vec<L1DaRef> },
+    /// Proving started, waiting for proof
+    ProofPending {
+        da: Vec<L1DaRef>,
```

Comment:
Do we need this to grow dynamically?

```suggestion
        da: &[L1DaRef],
```